src/Libs/GLib-2.0/Internal/SynchronizationContext.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace GLib.Internal;
+
+public sealed class GLibSynchronizationContext : SynchronizationContext
+{
+    #region Fields
+
+    // We need a static variable here instead of a static method because we need
+    // to keep the delegate object to stay alive for the application lifetime. We
+    // don't need a GCHandle for the delegate here because a static is enough to
+    // keep the delegate alive. See https://stackoverflow.com/q/75223488/1829232
+    // for a similar case.
+    private static readonly SourceFunc IdleCallback = data =>
+    {
+        var gch = GCHandle.FromIntPtr(data);
+        var action = (Action)gch.Target!;
+
+        action();
+
+        return false;
+    };
+
+    // According to https://docs.gtk.org/glib/main-loop.html#memory-management-of-sources
+    // we should not free the data in the idle callback because it is stored in a GSource
+    // object, which is still alive when the idle callback is returned. Instead, use
+    // either g_source_remove() or GDestroyNotify.
+    private static readonly DestroyNotify DestroyCallback = data =>
+    {
+        GCHandle.FromIntPtr(data).Free();
+    };
+
+    #endregion
+
+    #region Methods
+
+    public override SynchronizationContext CreateCopy()
+        => new GLibSynchronizationContext();
+
+    public override void Post(SendOrPostCallback d, object? state)
+        => ScheduleAction(() => d(state));
```

Comment:
Shouldn't here be a try/catch/ block to avoid crashes/undefined behavior?

Reply:
Yeah I did not catch the exception here because we don't have any mechanism to do something with it and swallow is not good too because it will hide the problem from programmers.